scrc/dataset_creation/criticality_dataset_creator.py

```diff
@@ -290,101 +261,65 @@ def process_citations(self, df):
 
         citation_count_df = self.references_df.copy()
         # iterate over unique values in column 'counter'
         a = citation_frequencies_df['counter'].unique()
         citation_count_df['counter'] = 0
+
         for i in a:
             # set counter in citation_count_df where citation_frequencies_df has matching reference and count value
             temp_freq_df = citation_frequencies_df[citation_frequencies_df['counter'] == i]
             temp_list = temp_freq_df['bge_file_number'].astype(str).tolist()
-            file_number_match = citation_count_df.bge_file_number_short.astype(str).isin(temp_list)
-            citation_count_df.loc[file_number_match, 'counter'] = int(i)
+            match = citation_count_df.bge_file_number_short.astype(str).isin(temp_list)
+            citation_count_df.loc[match, 'counter'] = i
         return citation_count_df
 
-    def get_citations(self, citations_as_string):
-        """
-        extract for each bger all ruling citations
-        :param citations_as_string:         citations how they were found in text of bger
-        :return:                            dataframe with additional column 'ruling_citation'
-        """
-        self.counter = self.counter + 1
-        if int(self.counter) % 10000 == 0:
-            print("Processed another 10'000")
-        cits = []
-        try:
-            citations = ast.literal_eval(citations_as_string)  # parse dict string to dict again
-            for citation in citations:
-                try:
-                    cit = citation['text']
-                    citation_type = citation['name']
-                    cit = ' '.join(cit.split())  # remove multiple whitespaces inside
-                    if citation_type == "ruling":
-                        cited_file = self.get_file_number(cit)
-                        cits.append(cited_file)
-                    elif citation_type == "law":
-                        pass
-                    else:
-                        raise ValueError("type of citation must be 'rulings' or 'law")
-                except ValueError as ve:
-                    self.logger.info(f"Citation has invalid syntax: {citation}")
-                    continue
-        except ValueError as ve:
-            self.logger.info(f"Citations could not be extracted to dict: {citations_as_string}")
-        if cits:  # only return something if we actually have citations
-            return cits
-
-    def get_file_number(self, citation):
+    def build_tf_matrix(self, df):
+        if not self.count_all_cits:
+            # count citations for one bge only once in bger
+            df['ruling_citation'] = df['ruling_citation'].apply(lambda x: set(x))
+
+        # create counter out of list
+        df['ruling_citation'] = df['ruling_citation'].apply(lambda x: Counter(x))
+
+        # multiply weight to value in counter
+        def weight_citations(row):
+            weight = (row['year'] - 2001)
```

Comment:
Is it possible to replace this magic number with a constant or even compute it? (e.g., the lowest year in the df)